test/starlark_tests/targets_under_test/apple/BUILD

```diff
@@ -273,6 +275,141 @@ apple_xcframework(
     deps = [":fmwk_lib"],
 )
 
+# Generate importable dynamic XCFramework files
+genrule(
+    name = "generated_ios_dynamic_xcframework",
+    testonly = 1,
+    srcs = [":ios_dynamic_xcframework.xcframework.zip"],
+    outs = ["ios_dynamic_xcframework.xcframework/Info.plist"] + [
+        "{0}/{1}/{2}/{3}".format(
+            "ios_dynamic_xcframework.xcframework",
+            platform_id,
+            "ios_dynamic_xcframework.framework",
+            file,
+        )
+        for file in [
+            "Headers/ios_dynamic_xcframework.h",
+            "Headers/shared.h",
+            "Modules/module.modulemap",
+            "ios_dynamic_xcframework",
+            "Info.plist",
+        ]
+        for platform_id in [
+            "ios-arm64",
+            "ios-x86_64-simulator",
+        ]
+    ],
+    cmd = """
+unzip -qq $(execpath :ios_dynamic_xcframework.xcframework.zip) -d $(RULEDIR)
+""",
+)
+
+# Generate importable static XCFramework files with static archives
+genrule(
+    name = "generated_ios_static_xcframework",
+    testonly = 1,
+    srcs = [":ios_static_xcframework.xcframework.zip"],
+    outs = [
+        "{0}/{1}/{2}".format(
+            "ios_static_xcframework.xcframework",
+            platform_id,
+            file,
+        )
+        for file in [
+            "Headers/shared.h",
+            "ios_static_xcframework.apple_static_library_lipo.a",
+        ]
+        for platform_id in [
+            "ios-arm64",
+            "ios-arm64_x86_64-simulator",
+        ]
+    ],
+    cmd = """
+unzip -qq $(execpath :ios_static_xcframework.xcframework.zip) -d $(RULEDIR)
+""",
+)
+
+# Generate importable static XCFramework files with static archives and module
+# map
+genrule(
+    name = "generated_ios_static_xcframework_with_module_map",
+    testonly = 1,
+    srcs = [":ios_static_xcframework_with_module_map.xcframework.zip"],
+    outs = [
+        "{0}/{1}/{2}".format(
+            "ios_static_xcframework_with_module_map.xcframework",
+            platform_id,
+            file,
+        )
+        for file in [
+            "Headers/shared.h",
+            "Modules/module.modulemap",
+            "ios_static_xcframework_with_module_map.apple_static_library_lipo.a",
```

Comment:
if it's not too much to ask for now, can we add a test target similar to this one that uses a static framework?